### DISCUSSION TITLE
sc-im and Lua

### DISCUSSION THREAD

**Discussion by DarkWiiPlayer** [andmarti1424/sc-im#556] [General]
Looking through the code, I couldn't help but notice that, while sc-im technically has built-in Lua scripting, all the built-in features where using such a scripting language would usually be of use are still written in C. Is there a reason for this?

**Comment by andmarti1424** [comment]
what do you mean? they are in lua? did you see the example of script in the
lua directory?

**Reply by DarkWiiPlayer** [reply]
Yes; but I was asking specifically about the implementation of sc-im itself in the `src` directory

**Comment by andmarti1424** [comment]
I havent done it but I believe its the easiest way of merging it to the already stuff done in  C.

**Comment by LLLLTian** [comment]
Hello, i am running into an error while installing it, can you guys help solve it. Thanks very much!
Here is the error msg:

/sc-im-main/src/trigger.c:222: undefined reference to `doLuaTrigger_cell'

**Comment by alexandre-mbm** [comment]
I would to use Python! It is friendly and full. But I am seeing that using @ext turns the things ugly.

**Reply by DarkWiiPlayer** [reply]
What would be the point in adding python, when Lua is already built into the tool *and* is specifically meant to be an embeddable language? Seems a bit like using a chain-saw to hammer in a nail to be honest.

**Reply by alexandre-mbm** [reply]
Sorry my ignorance. I need to study Lua. And Python...
